sklearn/metrics/_plot/calibration_curve.py

```diff
@@ -0,0 +1,193 @@
+from .base import _check_classifer_response_method
+
+from ...utils import check_matplotlib_support
+from ...utils.validation import _deprecate_positional_args
+from ...base import is_classifier
+from ...calibration import calibration_curve
+
+
+class CalibrationDisplay:
+    """Calibration visualization.
+
+    It is recommend to use :func:`~sklearn.metrics.plot_calibration_curve`
+    to create a visualizer. All parameters are stored as attributes.
+
+    Read more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    -----------
+    prob_true : ndarray
+        The proportion of samples whose class is the positive class, in each
+        bin (fraction of positives).
+
+    prob_pred : ndarray
+        The mean predicted probability in each bin.
+
+    estimator_name : str, default=None
+        Name of estimator. If None, then the estimator name is not shown.
+
+    Attributes
+    ----------
+    line_ : matplotlib Artist
+        Calibration curve.
+
+    ax_ : matplotlib Axes
+        Axes with calibration curve.
+
+    figure_ : matplotlib Figure
+        Figure containing the curve.
+    """
+    def __init__(self, prob_true, prob_pred, *,
+                 estimator_name=None):
+        self.prob_true = prob_true
+        self.prob_pred = prob_pred
+        self.estimator_name = estimator_name
+
+    @_deprecate_positional_args
```

Comment:
We would need this deprecation since this is a new class and new function?

Reply:
Should I move the `*` to not allow any positional args..? (bit confused about this)

Reply:
Specifically here `def plot(self, ax=None, *, name=None, ref_line=True, **kwargs):` as `ax` arguably should be keyword.